### PR TITLE
feature: 3D chessboard story

### PR DESCRIPTION
**Pull request by Manukyanq**
This PR is more about visualization and custom styling, by using `react-chessboard` styling features. 

image

### CODE REVIEW

File: src/chessboard/components/Piece.tsx [hunk 1]
```diff
@@ -118,13 +118,15 @@ export function Piece({
...
       if (sourceSq && targetSq) {
+        const squareWidth = boardWidth / 8;
         setPieceStyle((oldPieceStyle) => ({
           ...oldPieceStyle,
-          transform: `translate(${targetSq.x - sourceSq.x}px, ${
-            targetSq.y - sourceSq.y
-          }px)`,
+          transform: `translate(${
+            (targetSq.charCodeAt(0) - sourceSq.charCodeAt(0)) * squareWidth
+          }px, ${(Number(sourceSq[1]) - Number(targetSq[1])) * squareWidth}px)`,
```
Comment (Clariity) [inline src/chessboard/components/Piece.tsx:129]: why did this need to be changed?

Comment (Manukyanq) [reply src/chessboard/components/Piece.tsx:129]: `getBoundingClientRect()` does not calculate properly the {x,y} coordinates when we add  `transform: rotateX(27deg)`. The animation of piece movements becomes not correct

Comment (Manukyanq) [reply src/chessboard/components/Piece.tsx:129]: See how black pieceses move without these changes


https://github.com/Clariity/react-chessboard/assets/42891870/8437df2f-0aae-4a4a-a278-760b6e3d7fff

File: src/chessboard/components/Board.tsx [hunk 2]
```diff
@@ -33,32 +34,47 @@ export function Board() {
   }, []);
 
   return boardWidth ? (
-    <div ref={boardRef} style={{ position: "relative" }}>
-      <Squares />
-      <Arrows />
+    <div style={{ perspective: "1000px" }}>
```
Comment (Clariity) [inline src/chessboard/components/Board.tsx:37]: I've not used this property before, how does this being here not affect every board?

Comment (Manukyanq) [reply src/chessboard/components/Board.tsx:37]: CSS perspective property does not affect 2D elements. The perspective property is used in conjunction with 3D transformations in CSS to create a 3D perspective effect. It doesn't have any impact on purely 2D elements or transformations.

**Review by Clariity — APPROVED** [review]

**Pull request merged**